shm/Map.cpp

```diff
@@ -1,9 +1,21 @@
 #include "Map.hpp"
 
+#include <algorithm>
+
 void Map::setCurrentPosition(Island* const currentPosition) {
     currentPosition_ = currentPosition;
 }
 
 void Map::addIsland(const Island& newIsland) {
     islandsLocations_.push_back(newIsland);
 }
+
+Island* Map::getIsland(const Coordinates& coordinate) {
+    auto island = std::find_if(islandsLocations_.begin(), islandsLocations_.end(),
+                               [&coordinate](auto& isld) {
+                                   return isld.Island::getPosition() == coordinate;
```

Comment:
~~If you want to update position of an Island create a setter. Getters shouldn't be used to set variables.~~

Scratch that, you are comparing. I read that wrong